### PR TITLE
[GSS closed loop] BZ-2182943  FSGroup is not correctly set on subPath volume for CephFS CSI

### PR DESCRIPTION
**Pull request by Shrivaibavi**

### CODE REVIEW

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        )
+        # Check the owner group permissions
+        cmd_output = pod_obj.exec_cmd_on_pod(command="ls -l /etc/healing-controller.d/")
+        log.info(cmd_output)
+        cmd_output = cmd_output.split()
+        assert "root" in cmd_output[4] and cmd_output[13], "Owner is not set to root "
+        assert (
+            "9999" in cmd_output[5] and cmd_output[14]
+        ), "Owner group is not set to 9999"
+        log.info("FSGroup is correctly set on subPath volume for CephFS CSI ")
```
Comment (PrasadDesala) [inline tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py:100]: Can we also add an additional step at the end to restart the app pod and ensure that the owner group permissions are preserved?

File: ocs_ci/helpers/helpers.py [hunk 2]
```diff
@@ -225,7 +225,7 @@ def create_pod(
     interface_type=None,
     pvc_name=None,
     do_reload=True,
-    namespace=config.ENV_DATA["cluster_namespace"],
+    namespace=None,
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply ocs_ci/helpers/helpers.py]: Changed

File: ocs_ci/helpers/helpers.py [hunk 3]
```diff
@@ -267,7 +269,9 @@ def create_pod(
         ports (dict): Service ports
         deploy_pod_status (str): Expected status of deploy pod. Applicable
             only if dc_deployment is True
-        subpath (str): Value of subPath parameter in pod yaml
+        subpath (list): Value of subPath parameter in pod yaml
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply ocs_ci/helpers/helpers.py]: Changed

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+)
+from ocs_ci.ocs.ocp import OCP
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@polarion_id("OCS-4931")
+@bugzilla("2182943")
+@skipif_ocs_version("<4.12")
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py]: Done

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                "name": "mypvc",
+            },
+        ]
+        # Create pod with all the above security context and user/group permissions
+        pod_obj = create_pod(
+            namespace=project.namespace,
+            pvc_name=pvc.name,
+            interface_type=constants.CEPHFILESYSTEM,
+            security_context=security_context,
+            replica_count=1,
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py]: Its not a DC function, removed it

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py (hunk 1 above)
Comment (Shrivaibavi) [reply tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py]: Incorporated.

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            pvc.delete()
+            project.delete(resource_name=project.namespace)
+
+        request.addfinalizer(finalizer)
+
+        # Create project and pvc
+        project = create_project()
+        pvc = create_pvc(
+            sc_name=constants.CEPHFILESYSTEM_SC, namespace=project.namespace
+        )
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py]: Using pvc_factory instead

File: ocs_ci/helpers/helpers.py [hunk 7]
```diff
@@ -270,6 +272,8 @@ def create_pod(
             only if dc_deployment is True
         subpath (str): Value of subPath parameter in pod yaml
         deployment (bool): True for Deployment creation, False otherwise
+        scc (dict): Security context for fsGroup, runAsUer, runAsGroup
```
Comment (jilju) [inline ocs_ci/helpers/helpers.py:275]: There is another parameter "security_context" in this function which is set for the container. Please mention that "scc" is for the pod . OR please update the description of "security_context" to mention that it is sent on container.

File: tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        pod.ocp.wait_for_delete(resource_name=pod.name)
+        log.info("Creating pod and mounting the same PVC")
+        pod.create()
+        wait_for_resource_state(
+            resource=pod, state=constants.STATUS_RUNNING, timeout=300
+        )
+
+        validate_permissions(pod)
+
+        teardown_factory(pod)
```
Comment (jilju) [inline tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py:110]: Add this immediately after creating pod. ie, after line 89 in this commit.

Comment (Shrivaibavi) [reply tests/manage/pv_services/test_cephfs_pvc_with_necessary_scc_permissions.py]: done

**Review by jilju — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**